parsons/redash/redash.py

```diff
@@ -0,0 +1,186 @@
+    `Returns:`
+        Redash Class
+    """
+
+    def __init__(self,
+                 base_url=None,
+                 user_api_key=None,
+                 pause_time=3,
+                 timeout=0,  # never timeout
+                 verify=True):
+        self.base_url = check('REDASH_BASE_URL', base_url)
+        self.user_api_key = check('REDASH_USER_API_KEY', user_api_key, optional=True)
+        self.pause = int(check('REDASH_PAUSE_TIME', pause_time, optional=True))
+        self.timeout = int(check('REDASH_TIMEOUT', timeout, optional=True))
+
+        self.verify = verify  # for https requests
+        self.session = requests.Session()
+        if user_api_key:
+            self.session.headers.update({'Authorization': f'Key {user_api_key}'})
+
+    def _poll_job(self, session, job, query_id):
+        start_secs = time.time()
+        while job['status'] not in (3, 4):
+            if self.timeout and start_secs + self.timeout < time.time():
+                raise RedashTimeout(f'Redash timeout: {self.timeout}')
+            poll_url = '{}/api/jobs/{}'.format(self.base_url, job['id'])
+            response = session.get(poll_url, verify=self.verify)
+            response_json = response.json()
+            job = response_json.get(
+                'job',
+                {'status': 'Error NO JOB IN RESPONSE: {}'.format(json.dumps(response_json))})
+            logger.debug('poll', poll_url, query_id, job['status'], job.get('error'))
+            time.sleep(self.pause)
+
+        if job['status'] == 3:  # 3 = completed
+            return job['query_result_id']
+        elif job['status'] == 4:  # 3 = ERROR
+            raise RedashQueryFailed('Redash Query {} failed: {}'.format(query_id, job['error']))
+
+    def get_fresh_query_results(self, query_id=None, params=None):
+        """
+        Make a fresh query result and get back the CSV http response object back
+        with the CSV string in result.content
+
+        `Args:`
+            query_id: str or int
+                The query id of the query
+            params: dict
+                If there are values for the redash query parameters
+                (described https://redash.io/help/user-guide/querying/query-parameters
+                e.g. "{{datelimit}}" in the query),
+                then this is a dict that will pass the parameters in the POST.
+                We add the "p_" prefix for parameters, so if your query had ?p_datelimit=....
+                in the url, you should just set 'datelimit' in params here.
+                If you set this with REDASH_QUERY_PARAMS environment variable instead of passing
+                the values, then you must include the "p_" prefixes and it should be a single
+                url-encoded string as you would see it in the URL bar.
+        `Returns:`
+            Table Class
+        """
+        query_id = check('REDASH_QUERY_ID', query_id, optional=True)
```

Comment:
This makes sense to do here, because this is sometimes called by the `load_to_table` classmethod below.. but for future reference, usually the environment variable pattern is more about `__init__` parameters than necessarily specific function parameters.